### PR TITLE
RavenDB-13553 Changing backup from Backup to Snapshot should clear the backup status

### PR DESCRIPTION
**Pull request by YonatanResh**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-13553/Changing-backup-from-Backup-to-Snapshot-should-clear-the-backup-status

_Please delete below the options that are not relevant_

### Type of change

- Bug fix

### How risky is the change?

- Low 

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- Tests have been added that prove the fix is effective or that the feature works
- It has been verified by manual testing

### CODE REVIEW

File: src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs [hunk 1]
```diff
@@ -780,6 +780,11 @@ public void UpdateConfigurations(DatabaseRecord databaseRecord)
             var previousConfiguration = existingBackupState.Configuration;
             existingBackupState.Configuration = newConfiguration;
 
+            if (previousConfiguration.BackupType != newConfiguration.BackupType)
+            {
```
Comment (grisha-kotler) [inline src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs:784]: Please add a comment that we clear the in-memory state here

File: test/SlowTests/Issues/RavenDB-13553.cs [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                {
+                    session.Store(new User
+                    {
+                        Name = "Yonatan"
+                    });
+
+                    session.SaveChanges();
+                }
+
+                var config = new PeriodicBackupConfiguration
```
Comment (grisha-kotler) [inline test/SlowTests/Issues/RavenDB-13553.cs]: Use `Backup.CreateBackupConfiguration`

File: test/SlowTests/Issues/RavenDB-13553.cs [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                    BackupType = BackupType.Backup,
+
+                    Name = "fullBackupTask",
+                };
+
+                var operation = new UpdatePeriodicBackupOperation(config);
+                var result = await store.Maintenance.SendAsync(operation);
+
+                var backupStatus = await store.Maintenance.SendAsync(new StartBackupOperation(true, result.TaskId));
+                var x = await backupStatus.WaitForCompletionAsync(TimeSpan.FromMinutes(5));
```
Comment (grisha-kotler) [inline test/SlowTests/Issues/RavenDB-13553.cs]:
```suggestion
                await backupStatus.WaitForCompletionAsync(TimeSpan.FromMinutes(5));
```

File: test/SlowTests/Issues/RavenDB-13553.cs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+                config.BackupType = BackupType.Snapshot;
+                config.TaskId = result.TaskId;
+
+                operation = new UpdatePeriodicBackupOperation(config);
+                var result2 = await store.Maintenance.SendAsync(operation);
```
Comment (grisha-kotler) [inline test/SlowTests/Issues/RavenDB-13553.cs]: What do you do with `result2`?

File: test/SlowTests/Issues/RavenDB-13553.cs [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+                var documentDatabase = await Databases.GetDocumentDatabaseInstanceFor(store);
+                documentDatabase.PeriodicBackupRunner.ForTestingPurposesOnly().SimulateFailedBackup = true;
+
+                await Assert.ThrowsAsync<Raven.Client.Exceptions.RavenException>(async () =>
+                {
+                    var backupStatus2 = await store.Maintenance.SendAsync(new StartBackupOperation(true, result2.TaskId));
```
Comment (grisha-kotler) [inline test/SlowTests/Issues/RavenDB-13553.cs]: User `Backup.RunBackupAsync`

**Review by grisha-kotler — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs (hunk 1 above)
Comment (YonatanResh) [reply src/Raven.Server/Documents/PeriodicBackup/PeriodicBackupRunner.cs:784]: done

File: test/SlowTests/Issues/RavenDB-13553.cs (hunk 2 above)
Comment (YonatanResh) [reply test/SlowTests/Issues/RavenDB-13553.cs]: done

File: test/SlowTests/Issues/RavenDB-13553.cs (hunk 5 above)
Comment (YonatanResh) [reply test/SlowTests/Issues/RavenDB-13553.cs]: done

File: test/SlowTests/Issues/RavenDB-13553.cs (hunk 4 above)
Comment (YonatanResh) [reply test/SlowTests/Issues/RavenDB-13553.cs]: you are right, we dont need it, I removed it

File: test/SlowTests/Issues/RavenDB-13553.cs (hunk 3 above)
Comment (YonatanResh) [reply test/SlowTests/Issues/RavenDB-13553.cs]: I deleted that line, we didn't needed it anymore

**Review by grisha-kotler — APPROVED** [review]

**Pull request merged**